src/interfaces/node.cpp

```diff
@@ -221,15 +219,15 @@ class NodeImpl : public Node
     }
     std::vector<std::unique_ptr<Wallet>> getWallets() override
     {
-#ifdef ENABLE_WALLET
+        if (!g_wallet_init_interface.HasWalletSupport()) {
+            throw std::logic_error("Node::getWallets() called in non-wallet build.");
```

Comment:
Can you drop g_wallet_init_interface references and throws in this file? They seem redundant if MakeWallet and GetWallets functions can throw now.

Reply:
Done!